sstables/sstable_set.cc

```diff
@@ -390,10 +390,15 @@ void time_series_sstable_set::for_each_sstable(std::function<void(const shared_s
 
 // O(log n)
 void time_series_sstable_set::insert(shared_sstable sst) {
+  try {
     auto min_pos = sst->min_position();
     auto max_pos_reversed = sst->max_position().reversed();
     _sstables->emplace(std::move(min_pos), sst);
     _sstables_reversed->emplace(std::move(max_pos_reversed), std::move(sst));
+  } catch (...) {
+    erase(sst);
+    throw;
+  }
```

Comment:
Should this have a Fixes #? It looks like an unrelated bug fix.

Reply:
Yes. (though I don't if this was ever hit)

Reply:
I opened https://github.com/scylladb/scylla/issues/10787
I'll rework the commitlog in the next version.